src/sentry/integrations/mixins/notifications.py

```diff
@@ -26,10 +25,3 @@ def notify_remove_external_team(self, external_team: ExternalActor, team: Team)
             channel_id=external_team.external_id,
             message=SUCCESS_UNLINKED_TEAM_MESSAGE.format(team=team.slug),
         )
-
-    def remove_notification_settings(self, actor_id: int, provider: ExternalProviders) -> None:
-        """
-        Delete notification settings based on an actor_id
-        There is no foreign key relationship so we have to manually cascade.
-        """
-        NotificationSetting.objects._filter(target_ids=[actor_id], provider=provider).delete()
```

Comment:
Was this entirely unused?

Reply:
`NotificationsService.remove_notification_settings` is the method that is currently used in `ExternalActor.delete`.

Besides that, no known uses of `NotifyBasicMixin.remove_notification_settings` in sentry or getsentry.